src/js/request_scheduler.ts

```diff
@@ -16,7 +16,7 @@ export interface IResponse<T> {
 export interface IRequestScheduler {
     scheduleToPromise<T>(
         query: string,
-        event_converter: (evt: any) => any,
+        event_converter: (evt: { target: { responseText: string } }) => any,
```

Comment:
Are you making this change as part of "general improvements to crappy code" or because it's necessary to be able to extract and display category.
I'm down for either (and seeing a naked 'any' littered about is an admission of crappy code if ever there was one), but probably not in the same pull request.
If the former (general improvement to typing), would you mind breaking it out into another PR that doesn't break the "secret" test pack I have with user data embedded into it?

As an advance review on that code offering, do you think we should declare and name a type, such as AzadEvent or similar and use that instead of duplicating it across all _schedule calls?